src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -234,16 +266,15 @@ protected void runTool(final JavaSparkContext ctx) {
      *
      * This will be replaced by a parallel writer similar to what's done with {@link org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSink}
      */
-    private void writeVariants(JavaRDD<VariantContext> variants, HaplotypeCallerEngine hcEngine) {
+    private static void writeVariants(String outputFile, JavaRDD<VariantContext> variants, HaplotypeCallerEngine hcEngine, SAMSequenceDictionary referenceDictionary, SAMSequenceDictionary readsDictionary) {
         final List<VariantContext> collectedVariants = variants.collect();
-        final SAMSequenceDictionary referenceDictionary = getReferenceSequenceDictionary();
 
         final List<VariantContext> sortedVariants = collectedVariants.stream()
             .sorted((o1, o2) -> IntervalUtils.compareLocatables(o1, o2, referenceDictionary))
             .collect(Collectors.toList());
 
-        try(final VariantContextWriter writer = hcEngine.makeVCFWriter(output, getBestAvailableSequenceDictionary())) {
-            hcEngine.writeHeader(writer, getHeaderForReads().getSequenceDictionary(), new HashSet<>());
+        try(final VariantContextWriter writer = hcEngine.makeVCFWriter(outputFile, referenceDictionary)) {
+            hcEngine.writeHeader(writer, readsDictionary, new HashSet<>());
             sortedVariants.forEach(writer::add);
         }
```

Comment:
Could you remove the now unused method `getShardBoundaryOverlapDetector` on line 286?  It became dead code in a previous PR but I didn't notice until now.